Remotion/Data/DomainObjects/Mapping/TypeDefinition.cs

```diff
@@ -180,14 +186,16 @@ public void SetPropertyDefinitions (PropertyDefinitionCollection propertyDefinit
 
     public RelationEndPointDefinitionCollection GetRelationEndPointDefinitions ()
     {
-      return _cachedRelationEndPointDefinitions.Value;
+      return IsReadOnly
+        ? _cachedRelationEndPointDefinitions.Value
+        : CreateRelationEndPointDefinitionCollection();
     }
 
     public IRelationEndPointDefinition? GetRelationEndPointDefinition (string propertyName)
     {
       ArgumentUtility.CheckNotNullOrEmpty(nameof(propertyName), propertyName);
 
-      return _cachedRelationEndPointDefinitions.Value[propertyName];
+      return GetRelationEndPointDefinitions()[propertyName];
```

Comment:
Here and GetPropertyDefinition: since you're touching those lines and changing it to a method call: do we have a method avaiable instead of the indexer? It reads better.